pkg/objstore/s3/s3.go

```diff
@@ -184,26 +184,43 @@ func (b *Bucket) Iter(ctx context.Context, dir string, f func(string) error) err
 	return nil
 }
 
+func (b *Bucket) getRange(ctx context.Context, name string, off, length int64) (io.ReadCloser, error) {
+	b.opsTotal.WithLabelValues(opObjectGet).Inc()
+	opts := &minio.GetObjectOptions{ServerSideEncryption: b.sse}
+	if length != -1 {
+		if err := opts.SetRange(off, off+length-1); err != nil {
```

Comment:
I don't have enough context, but what if off+length is bigger than total bucket size?

Reply:
it should be handled by lib (since we are making client call only) but good point.